File: Week01/Decimal_to_Binary/HeadRecursion.cs
using System.Text;
using static System.Console;

class HeadRecursion
{
    static void decToBinaryRec(int n, StringBuilder bin)
    {
        if (n == 0)
            return;

        decToBinaryRec(n / 2, bin);

        bin.Append(n % 2);
    }

    static string decToBinary(int n)
    {
        if (n == 0)
            return "0";

        StringBuilder bin = new StringBuilder();
        decToBinaryRec(n, bin);
        return bin.ToString();
    }


    static void Main()
    {
        int n = 12;
        WriteLine(decToBinary(n));
    }
}

File: Week01/Decimal_to_Binary/HeadRecursion.cpp
#include<iostream>
#include<algorithm>

using namespace std;

void decToBinaryRec(int n, string &bin) {
    if (n == 0)
        return;
    
    decToBinaryRec(n/2, bin);

    bin.push_back(n%2 + '0');
}


string decToBinary(int n) {
    if (n == 0)
        return "0";
    
    string bin = "";
    decToBinaryRec(n, bin);
    return bin;
}


int main() {
    int n = 12;
    cout << decToBinary(n);
    return 0;
}